test/CMakeLists.txt

```diff
@@ -34,7 +34,7 @@ foreach(test ${tests})
   add_executable(${test} EXCLUDE_FROM_ALL ${test}.cc )
   set_target_properties(${test} PROPERTIES
     COMPILE_DEFINITIONS VALHALLA_SOURCE_DIR="${VALHALLA_SOURCE_DIR}/")
-  target_link_libraries(${test} valhalla gtest gmock)
+  target_link_libraries(${test} valhalla gtest gmock -lpthread)
```

Comment:
do you need the `-l` when using `target_link_libraries`? Alternatively, we can use the same style we use in the top-level `CMakeLists.txt`:
```cmake
target_link_libraries(${test} valhalla gtest gmock Threads::Threads)
```
(although I'm no 100% sure the above syntax is supported by our minimum cmake version (`3.5`).

Reply:
oh yeah now that you mention it i do remember seeing that! thanks

Reply:
sadly this didnt work. the best i could get to work was `pthread`